enterprise/coderd/prebuilds/reconcile.go

```diff
@@ -243,6 +243,11 @@ func (c *StoreReconciler) ReconcileAll(ctx context.Context) error {
 		var eg errgroup.Group
 		// Reconcile presets in parallel. Each preset in its own goroutine.
 		for _, preset := range snapshot.Presets {
+			if snapshot.IsHardLimited(preset.ID) {
+				logger.Debug(ctx, "skipping hard limited preset", slog.F("preset_id", preset.ID), slog.F("name", preset.Name))
```

Comment:
This should be higher than `debug`; it would be a `warn`, and we probably need something more than a log to inform the operator of this situation.

Can you add a metric as well? We'll need to have docs for how to work around this situation, too.

Presets in this state will get stuck forever until either a template version is pushed, right? Even deleting a prebuild won't help because I think we just soft-delete the workspace, so the build will still be present.

Reply:
I agree that we need either `metrics` or `coder notifications` or both. But I wanted to do it in separate pull request.

Regarding log message, I wasn't sure if we need a `warn` here because, we'll notify an admin via `coder notifications` anyway. But fix may take a while, and we'll print this warn message every 15 secs, is it expected behaviour?

Reply:
Why wait for another PR to add those? This PR is pretty small and we can link all the related pieces together.

Reply:
done